Bags-list tests don't compile

### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Description of bug

When running: ```cargo test -p pallet-bags-list``` the following error occurs: 
```
error[E0407]: method `set_score_of` is not a member of trait `frame_election_provider_support::ScoreProvider`
  --> frame/bags-list/src/mock.rs:44:2
   |
44 | /     fn set_score_of(id: &AccountId, weight: Self::Score) {
45 | |         NEXT_VOTE_WEIGHT_MAP.with(|m| m.borrow_mut().insert(*id, ...
46 | |     }
   | |_____^ not a member of trait `frame_election_provider_support::ScoreProvider`

warning: unused import: `frame_election_provider_support::ScoreProvider`
  --> frame/bags-list/src/benchmarks.rs:25:5
   |
25 | use frame_election_provider_support::ScoreProvider;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

error[E0599]: no function or associated item named `set_score_of` found for associated type `<T as Config<I>>::ScoreProvider` in the current scope
  --> frame/bags-list/src/benchmarks.rs:75:21
   |
75 |         T::ScoreProvider::set_score_of(&origin_middle, dest_bag_thresh);
   |                           ^^^^^^^^^^^^ function or associated item not found in `<T as Config<I>>::ScoreProvider`

error[E0599]: no function or associated item named `set_score_of` found for associated type `<T as Config<I>>::ScoreProvider` in the current scope
   --> frame/bags-list/src/benchmarks.rs:134:21
    |
134 |         T::ScoreProvider::set_score_of(&origin_tail, dest_bag_thresh);
    |                           ^^^^^^^^^^^^ function or associated item not found in `<T as Config<I>>::ScoreProvider`
.
.
.
And so on
```

### Steps to reproduce

```cargo test -p pallet-bags-list```

--- a/frame/bags-list/src/lib.rs
+++ b/frame/bags-list/src/lib.rs
@@ -397,7 +397,7 @@ impl<T: Config<I>, I: 'static> ScoreProvider<T::AccountId> for Pallet<T, I> {
 		Node::<T, I>::get(id).map(|node| node.score()).unwrap_or_default()
 	}
 
-	frame_election_provider_support::runtime_benchmarks_or_fuzz_enabled! {
+	frame_election_provider_support::runtime_benchmarks_fuzz_or_std_enabled! {
 		fn set_score_of(id: &T::AccountId, new_score: T::Score) {
 			ListNodes::<T, I>::mutate(id, |maybe_node| {
 				if let Some(node) = maybe_node.as_mut() {

--- a/frame/bags-list/src/mock.rs
+++ b/frame/bags-list/src/mock.rs
@@ -40,9 +40,10 @@ impl frame_election_provider_support::ScoreProvider<AccountId> for StakingMock {
 		*NextVoteWeightMap::get().get(id).unwrap_or(&NextVoteWeight::get())
 	}
 
-	#[cfg(any(feature = "runtime-benchmarks", feature = "fuzz", test))]
-	fn set_score_of(id: &AccountId, weight: Self::Score) {
-		NEXT_VOTE_WEIGHT_MAP.with(|m| m.borrow_mut().insert(*id, weight));
+	frame_election_provider_support::runtime_benchmarks_fuzz_or_std_enabled! {
+		fn set_score_of(id: &AccountId, weight: Self::Score) {
+			NEXT_VOTE_WEIGHT_MAP.with(|m| m.borrow_mut().insert(*id, weight));
+		}
 	}
 }
 

--- a/frame/election-provider-support/src/lib.rs
+++ b/frame/election-provider-support/src/lib.rs
@@ -582,7 +582,7 @@ pub trait ScoreProvider<AccountId> {
 	fn score(who: &AccountId) -> Self::Score;
 
 	/// For tests, benchmarks and fuzzing, set the `score`.
-	#[cfg(any(feature = "runtime-benchmarks", feature = "fuzz", test))]
+	#[cfg(any(feature = "runtime-benchmarks", feature = "fuzz", feature = "std"))]
 	fn set_score_of(_: &AccountId, _: Self::Score) {}
 }
 
@@ -673,5 +673,14 @@ pub type BoundedSupportsOf<E> = BoundedSupports<
 	<E as ElectionProviderBase>::MaxWinners,
 >;
 
-sp_core::generate_feature_enabled_macro!(runtime_benchmarks_enabled, feature = "runtime-benchmarks", $);
-sp_core::generate_feature_enabled_macro!(runtime_benchmarks_or_fuzz_enabled, any(feature = "runtime-benchmarks", feature = "fuzzing"), $);
+sp_core::generate_feature_enabled_macro!(
+	runtime_benchmarks_enabled,
+	feature = "runtime-benchmarks",
+	$
+);
+
+sp_core::generate_feature_enabled_macro!(
+	runtime_benchmarks_fuzz_or_std_enabled,
+	any(feature = "runtime-benchmarks", feature = "fuzzing", feature = "std"),
+	$
+);

